cmd/collector/app/processor/interface.go

```diff
@@ -28,6 +29,7 @@ var ErrBusy = errors.New("server busy")
 type SpansOptions struct {
 	SpanFormat       SpanFormat
 	InboundTransport InboundTransport
+	Context          context.Context
```

Comment:
have a concern about this one. First, it is not idiomatic and explicitly goes against recommendation to avoid storing context in structs. Second, the reason we have this struct is because we have a queue sitting between the RPC endpoint accepting spans and the processing pipeline. Which means the context coming from RPC has a different life-cycle than the processing life-cycle. For example, by the time the item is processed from the queue the RPC context may have been cancelled already, which won't affect passing the tenancy, but might affect sending DB requests if the same context is piped all the way through the writers.

I think it would be better to capture the tenancy explicitly at this boundary and let the queue consumer create its own context (and re-save the tenancy there if necessary)